apis/apiextensions/v1/composition_patches.go

```diff
@@ -216,6 +245,11 @@ func (c *Patch) applyFromFieldPathPatch(from, to runtime.Object) error {
 		return err
 	}
 
+	// Patch all expanded fields if the ToFieldPath contains wildcards
+	if strings.Contains(*c.ToFieldPath, "[*]") {
```

Comment:
[ExpandWildcards returns the same path](https://github.com/crossplane/crossplane-runtime/blob/edc27bfc3ed3c672c0cd5302c80ef5c49a6a7e81/pkg/fieldpath/paved_test.go#L844-L875) if no wildcard exists, so, we could always use the same code path without this check.

However, I am on the fence regarding performance. If this `strings.Contains` check works well for all cases, I agree that this is a good optimization.

Reply:
I didn't do any performance benchmarking, but my presumption was that using a check to avoid using the multiple patching would be more performant. Another factor that makes it good to do this is that the current implementation of `patchFieldValuetoMultiple` only works if the supplied field already exists in the defined MR unlike the implementation of `patchFieldvalueToObject`, which inserts a new field if it doesn't already exist.

Reply:
Performance isn't of particular concern here; we should optimize for simpler and more readable code. We can always tune for performance once we notice an issue.

Reply:
> Another factor that makes it good to do this is that the current implementation of patchFieldValuetoMultiple only works if the supplied field already exists in the defined MR unlike the implementation of patchFieldvalueToObject, which inserts a new field if it doesn't already exist.

Ok, noticed this as well. If we always used expand wildcards, we wouldn't be able to patch labels unless the object to be patched has already some labels.